clients/ts/signalr/tests/Common.ts

```diff
@@ -21,3 +22,27 @@ export function eachEndpointUrl(action: (givenUrl: string, expectedUrl: string)
 
     urls.forEach((t) => action(t[0], t[1]));
 }
+
+export class VerifyLogger implements ILogger {
+    public unexpectedErrors: string[];
+    private expectedErrors: string[];
+
+    public constructor(...expectedErrors: string[]) {
+        this.unexpectedErrors = new Array<string>();
+        this.expectedErrors = expectedErrors;
+    }
+
+    public static async run(fn: (logger: VerifyLogger) => Promise<any>, ...expectedErrors: string[]): Promise<any> {
+        const logger = new VerifyLogger(...expectedErrors);
+        await fn(logger);
+        expect(logger.unexpectedErrors.length).toBe(0);
+    }
+
+    public log(logLevel: LogLevel, message: string): void {
+        if (logLevel >= LogLevel.Error) {
+            if (this.expectedErrors.filter((error) => error === message).length === 0) {
```

Comment:
Add support for `RegExp`s here. So have the `expectedErrors` be `Array<RegExp | string>` and have this code check if `error instanceof RegExp` (either move the filter call out of the `if` or put all that in a helper method to avoid overloading this line)

Reply:
Maybe make it a `(string) => boolean` and support each item in the array being any of `string`, `RegExp` or `(string) => boolean` (and marshal instances of the first two into functions)